src/pages-and-resources/discussions/app-config-form/apps/shared/BlackoutDatesField.jsx

```diff
@@ -68,49 +70,62 @@ import messages from './messages';
  */
 export const blackoutDatesRegex = /^\[(\[("[0-9]{4}-(0[1-9]|1[0-2])-[0-3][0-9](T([0-1][0-9]|2[0-3]):([0-5][0-9])){0,1}"),("[0-9]{4}-(0[1-9]|1[0-2])-[0-3][0-9](T([0-1][0-9]|2[0-3]):([0-5][0-9])){0,1}")\](,){0,1})*\]$/;
 
-function BlackoutDatesField({
-  onBlur,
-  onChange,
-  intl,
-  values,
-  errors,
-}) {
+const BlackoutDatesField = ({ intl }) => {
+  const [inFocus, setInFocus] = useState(false);
+  const dispatch = useDispatch();
+  const {
+    handleChange, handleBlur, errors,
+    touched, values: appConfig,
+  } = useFormikContext();
+
+  const hasError = Boolean(touched.blackoutDates && errors.blackoutDates);
+
+  useEffect(() => {
+    dispatch(updateValidationStatus({ hasError }));
+  }, [hasError]);
```

Comment:
the same logic is repeated in topic items. I feel like we can have a generic solution for that

Reply:
Sure. We will discuss this point.